tests/integration/conftest.py

```diff
@@ -0,0 +1,170 @@
+scripts_dir = root_repo_dir / "scripts"
+complytime_cache_dir = Path("/tmp/trestle-bot-complytime-cache")
+complytime_cache_dir.mkdir(parents=True, exist_ok=True)
+int_test_data_dir = Path(__file__).parent.parent / "integration_data/"
+_TEST_PREFIX = "trestlebot_tests"
+
+T = TypeVar("T")
+YieldFixture = Generator[T, None, None]
+
+
+def is_complytime_installed(install_dir: Path) -> bool:
+    install_dir / ".config/complytime"
+    openscap_plugin_bin = (
+        install_dir / ".config/complytime/plugins/openscap-plugin"
+    ).resolve()
+    openscap_plugin_conf = (
+        install_dir / ".config/complytime/plugins/c2p-openscap-manifest.json"
+    ).resolve()
+    if not openscap_plugin_bin.exists():
+        return False
+    if not openscap_plugin_conf.exists():
+        return False
+    return True
+
+
+def is_complytime_cached(download_dir: Path) -> bool:
+    return bool(
+        glob.glob(
+            str((download_dir / "releases/*/complytime_linux_x86_64.tar.gz").resolve())
+        )
+    )
+
+
+def sha256sum(filepath: Path) -> str:
+    sha256 = hashlib.sha256()
+    chunk_size = 65536
+    with open(filepath, "rb") as file:
+        while True:
+            chunk = file.read(chunk_size)
+            if not chunk:
+                break
+            sha256.update(chunk)
+        return sha256.hexdigest()
+
+
+@pytest.fixture(autouse=True)
+def complytime_home() -> YieldFixture[Path]:
+    # Setup
+    complytime_cache_dir.mkdir(parents=True, exist_ok=True)
+    complytime_home = Path(tempfile.mkdtemp(prefix=_TEST_PREFIX))
+    orig_home = os.getenv("HOME")
+    orig_path = os.getenv("PATH")
+    orig_xdg_config_home = os.getenv("XDG_CONFIG_HOME")
+
+    complytime_home.mkdir(parents=True, exist_ok=True)
+    if not is_complytime_installed(complytime_home):
+        if not is_complytime_cached(complytime_cache_dir):
+            result = subprocess.run(
+                [scripts_dir / "get-github-release.sh", "--prerelease"],
+                cwd=complytime_cache_dir,
+                capture_output=True,
+                text=True,
+            )
+            if result.returncode != 0:
+                raise ValueError(
+                    f"Unable to install ComplyTime for int test!\n{result.stdout}\n{result.stderr}"
+                )
+        result = subprocess.run(
+            [
+                "find",
+                f"{complytime_cache_dir}/releases",
+                "-name",
+                "complytime_linux_x86_64.tar.gz",
+                "-exec",
+                "tar",
+                "-xvf",
+                "{}",
+                ";",
+                "-quit",
+            ],
+            cwd=complytime_home,
+            capture_output=True,
+            text=True,
+        )
+        if result.returncode != 0:
+            raise ValueError(
+                f"Unable to extract ComplyTime for int test!\n{result.stdout}\n{result.stderr}"
+            )
+
+        # Install complytime
```

Comment:
Maybe this part could be moved to a new function? It could improve the readability and be easier to maintain.